blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java

```diff
@@ -95,72 +110,64 @@ public void startServer() {
-                            ctx.getChannel().write(
-                                    new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_REQUEST))
-                                    .addListener(ChannelFutureListener.CLOSE);
-                        } else {
-                            log.debug("channel " + ctx.getChannel().toString() + " is no longer writeable, not sending 400 response back to client");
-                        }
-                    } catch (Exception ex) {
-                        // If we are getting exception trying to write,
-                        // don't propagate to caller. It may cause this
-                        // method to be called again and will produce
-                        // stack overflow. So just log it here.
-                        log.debug("Can't write to channel " + ctx.getChannel().toString(), ex);
+            // if something bad happens during the decode, assume the client send bad data. return a 400.
+            @Override
+            public void exceptionCaught(ChannelHandlerContext ctx, Throwable thr) throws Exception {
+                try {
+                    if (ctx.channel().isWritable()) {
+                        log.debug("request decoder error " + thr.getCause().toString() + " on channel " + ctx.channel().toString());
+                        ctx.channel().write(
+                                new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_REQUEST))
+                                .addListener(ChannelFutureListener.CLOSE);
+                    } else {
+                        log.debug("channel " + ctx.channel().toString() + " is no longer writeable, not sending 400 response back to client");
                     }
+                } catch (Exception ex) {
+                    // If we are getting exception trying to write,
+                    // don't propagate to caller. It may cause this
+                    // method to be called again and will produce
+                    // stack overflow. So just log it here.
+                    log.debug("Can't write to channel " + ctx.channel().toString(), ex);
                 }
-            });
-            pipeline.addLast("chunkaggregator", new HttpChunkAggregator(MAX_CONTENT_LENGTH));
-            pipeline.addLast("inflater", new HttpContentDecompressor());
-            pipeline.addLast("encoder", new HttpResponseEncoder());
-            pipeline.addLast("encoder2", new HttpResponseDecoder());
-            pipeline.addLast("handler", new QueryStringDecoderAndRouter(router));
-
-            return pipeline;
-        }
+            }
+        });
+        pipeline.addLast("inflater", new HttpContentDecompressor());
+        pipeline.addLast("chunkaggregator", new HttpObjectAggregator(Constants.MAX_CONTENT_LENGTH));
+        pipeline.addLast("encoder2", new HttpResponseDecoder());
```

Comment:
I can change it to be respdecoder. There is another encoder and decoder up on top.